src/it/test-projects/creedengo-java-plugin-test-project/src/main/java/org/greencodeinitiative/creedengo/java/checks/AvoidRuntimeExceptions.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * creedengo - Java language - Provides rules to reduce the environmental footprint of your Java programs
+ * Copyright © 2024 Green Code Initiative (https://green-code-initiative.org/)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.greencodeinitiative.creedengo.java.checks;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.RuntimeException;
+import java.lang.ArithmeticException;
+import java.lang.IndexOutOfBoundsException;
+import java.lang.NullPointerException;
+import java.lang.IllegalArgumentException;
+import java.lang.NumberFormatException;
+import java.lang.Integer;
+
+public class AvoidRuntimeExceptions {
+    public void nominalRuntimeException() {
+        try {
+            // some code that may throw an exception
+        } catch (RuntimeException e) { // Noncompliant {{Avoid Runtime exceptions : RuntimeException}}
+        }
+    }
+
+    public void nominalRuntimeExceptionDependant() {
+        int[] array = new int[10];
+        try {
+            array[10] = 0;
+        } catch (IndexOutOfBoundsException e) { // Noncompliant {{Avoid Runtime exceptions : IndexOutOfBoundsException}}
+        }
+    }
+
+    public void nominalRuntimeExceptionDependant_2(){
+        Object obj = null;
+        try {
+            obj.toString();
+        } catch (NullPointerException e) { // Noncompliant {{Avoid Runtime exceptions : NullPointerException}}
+        }
+    }
+
+    public void nominalRuntimeExceptionDependant_3(){
+        try {
+            int result = 1 / 0;
+        } catch (ArithmeticException e) { // Noncompliant {{Avoid Runtime exceptions : ArithmeticException}}
+        }
+    }
+
+    //these exceptions are ok because they aren't runtime exceptions
+
+    public void compliantNonRuntimeException() {
```

Comment:
this use case is wrong ... FileNotFoundExceptinoand IOException aren't RuntimeException. please delete it.
please apply this feedback to other files

Reply:
hi @max-208 ... you're rigth ... sorry and read too quickly your PR. I completely agree with you.